src/web/app/pages-instructor/instructor-courses-page/instructor-courses-page.component.ts

```diff
@@ -216,18 +208,83 @@ export class InstructorCoursesPageComponent implements OnInit {
     this.courseService.changeArchiveStatus(courseId, {
       archiveStatus: toArchive,
     }).subscribe((courseArchive: CourseArchive) => {
-      this.loadInstructorCourses();
       if (courseArchive.isArchived) {
+        this.changeModelFromActiveToArchived(courseId);
         this.statusMessageService.showSuccessMessage(`The course ${courseId} has been archived.
           It will not appear on the home page anymore.`);
       } else {
+        this.changeModelFromArchivedToActive(courseId);
         this.statusMessageService.showSuccessMessage('The course has been unarchived.');
       }
     }, (resp: ErrorMessageOutput) => {
       this.statusMessageService.showErrorMessage(resp.error.message);
     });
   }
 
+  /**
+   * Moves a course model from active courses list to archived list.
+   * This is to reduce the need to refresh the entire list of courses multiple times.
+   */
+  changeModelFromActiveToArchived(courseId: string): void {
+    const courseToBeRemoved: CourseModel | undefined = this.findAndRemoveActiveCourse(courseId);
+    if (courseToBeRemoved !== undefined) {
+      this.archivedCourses.push(courseToBeRemoved);
+    }
+  }
+
+  /**
+   * Moves a course model from archived courses list to active list.
+   * This is to reduce the need to refresh the entire list of courses multiple times.
+   */
+  changeModelFromArchivedToActive(courseId: string): void {
+    const courseToBeRemoved: CourseModel | undefined = this.findAndRemoveArchivedCourse(courseId);
+    if (courseToBeRemoved !== undefined) {
+      this.activeCourses.push(courseToBeRemoved);
+      this.activeCourses.sort(this.sortBy(this.tableSortBy));
+    }
+  }
+
+  /**
+   * Finds and removes an active course from the active course list.
+   */
+  findAndRemoveActiveCourse(courseId: string): CourseModel | undefined {
```

Comment:
You can try to combine `findAndRemoveActiveCourse`. `,indAndRemoveArchivedCourse` and `findAndRemoveSoftDeletedCourse` to `findAndRemoveCourseModel(from, couseId)`?

Reply:
I tried setting the course model array as the parameter `from`. But the way used to remove the course by `filter` and assignment does not work because JavaScript makes a copy of the array. I also tried combining them into switch case statements but that makes the function lengthy. Any suggestions on the improvement?

Reply:
Try
```javascript
this.activeCourses = findAndRemoveCourseModel(findAndRemoveCourseModel, courseId)
```
?